### PR TITLE
Remove ignition

### PR DESCRIPTION
**Pull request by nkoenig**
# 🎉 New feature

Remove ignition

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/MainWindow.cc [hunk 1]
```diff
@@ -947,16 +947,7 @@ void MainWindow::SetShowDialogOnExit(bool _showDialogOnExit)
...
-  auto renderEngine = _renderEngine;
-  auto pos = renderEngine.find("ignition");
-  if (pos != std::string::npos)
-  {
-    renderEngine.replace(pos, pos + 8, "gz");
-    gzwarn << "Trying to load deprecated plugin [" << _renderEngine
-           << "]. Use [" << renderEngine << "] instead." << std::endl;
-  }
-  this->setProperty("renderEngine", renderEngine.c_str());
+ this->setProperty("renderEngine", _renderEngine.c_str());
```
Comment (ahcorde) [inline src/MainWindow.cc:950]:
```suggestion
  this->setProperty("renderEngine", _renderEngine.c_str());
```

**Review by jennuine — COMMENTED** [review]
I found a few more places:

* https://github.com/gazebosim/gz-gui/blob/main/tutorials/01_install.md

* https://github.com/gazebosim/gz-gui/blob/5bbfc6306cb6466aa659d6325d84311a06470ed8/include/gz/gui/qml/qmldir#L9-L11

* https://github.com/gazebosim/gz-gui/blob/5bbfc6306cb6466aa659d6325d84311a06470ed8/examples/plugin/gz_components/GzComponents.qml#L43-L50

* Are we leaving this? 
https://github.com/gazebosim/gz-gui/blob/6851f0453613fd8079e8c3378a7d94d3d847fcf8/src/Application.cc#L61-L63

* https://github.com/gazebosim/gz-gui/blob/6851f0453613fd8079e8c3378a7d94d3d847fcf8/src/Plugin.cc#L75

* https://github.com/gazebosim/gz-gui/blob/5bbfc6306cb6466aa659d6325d84311a06470ed8/src/plugins/marker_manager/MarkerManager.cc#L356

**Comment by nkoenig** [comment]
> 5844641



> I found a few more places:
> 
> * https://github.com/gazebosim/gz-gui/blob/main/tutorials/01_install.md
> * https://github.com/gazebosim/gz-gui/blob/5bbfc6306cb6466aa659d6325d84311a06470ed8/include/gz/gui/qml/qmldir#L9-L11
> * https://github.com/gazebosim/gz-gui/blob/5bbfc6306cb6466aa659d6325d84311a06470ed8/examples/plugin/gz_components/GzComponents.qml#L43-L50
> * Are we leaving this?
>   https://github.com/gazebosim/gz-gui/blob/6851f0453613fd8079e8c3378a7d94d3d847fcf8/src/Application.cc#L61-L63
> * https://github.com/gazebosim/gz-gui/blob/6851f0453613fd8079e8c3378a7d94d3d847fcf8/src/Plugin.cc#L75
> * https://github.com/gazebosim/gz-gui/blob/5bbfc6306cb6466aa659d6325d84311a06470ed8/src/plugins/marker_manager/MarkerManager.cc#L356

Remvoed in [5844641](https://github.com/gazebosim/gz-gui/pull/556/commits/5844641a444195bfbaaf0b11d41bbd340c13f6a3)

**Comment by mjcarroll** [comment]
I think there are still a few references to `pluginPathEnvDeprecated` judging by current CI

**Comment by iche033** [comment]
> I think there are still a few references to `pluginPathEnvDeprecated` judging by current CI

I think [5fbbcf2](https://github.com/gazebosim/gz-gui/pull/556/commits/5fbbcf2f9a5cfa41dd71d049de4f54c81de3f39f) should do it

**Review by jennuine — APPROVED** [review]
Thank you!

**Pull request merged**